File: core/src/main/java/com/yopox/screens/TitleScreen.java
package com.yopox.screens;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.graphics.Texture;
import com.yopox.DeviNUPES;
import com.yopox.SFX;
import com.yopox.Util;
import com.yopox.ui.Colors;
import com.yopox.ui.Letter;
import com.yopox.ui.Text;

import java.util.Vector;

public class TitleScreen extends AbstractScreen {
    private static final Texture background = new Texture("title.png");

    private final Vector<Letter> option1 = Letter.listOf("Partie du jour", Colors.DEFAULT_FG, Colors.DEFAULT_BG);
    private final Vector<Letter> option2 = Letter.listOf("5 minutes chrono", Colors.DEFAULT_FG, Colors.DEFAULT_BG);
    private final Vector<Letter> option3 = Letter.listOf("Mode duel", Colors.DEFAULT_FG, Colors.DEFAULT_BG);

    private final Vector<Letter> arrow = Letter.listOf(">", Colors.DEFAULT_FG, Colors.DEFAULT_BG);
    private int selected = 0;

    public TitleScreen(DeviNUPES game) {
        super(game);
    }

    @Override
    public void render(float delta) {
        super.render(delta);
        myBatch.begin();
        myBatch.draw(background, 0f, 0f);
        Text.draw(option1, Util.TILE * 13, Util.TILE * 8, false, myBatch);
        Text.draw(option2, Util.TILE * 13, Util.TILE * 6, false, myBatch);
        Text.draw(option3, Util.TILE * 13, Util.TILE * 4, false, myBatch);
        Text.draw(arrow, Util.TILE * 10, Util.TILE * 8 - selected * 2 * Util.TILE, false, myBatch);
        myBatch.end();
    }

    @Override
    public boolean keyUp(int keycode) {
        int n_OPTIONS = 3;
        switch (keycode) {
            case Input.Keys.UP:
                SFX.playKeySound();
                selected = (selected - 1 + n_OPTIONS) % n_OPTIONS;
                break;
            case Input.Keys.DOWN:
                SFX.playKeySound();
                selected = (selected + 1 + n_OPTIONS) % n_OPTIONS;
                break;
            default:
                if (selected == 0) {
                    SFX.OK.play(SFX.VOLUME);
                    myGame.startGame(Util.Mode.DAILY);
                } else if (selected == 1) {
                    SFX.OK.play(SFX.VOLUME);
                    myGame.startGame(Util.Mode.TIME_TRIAL);
                } else {
                    SFX.OK.play(SFX.VOLUME);
                    myGame.chooseSeed();
                }
                break;
        }
        return true;
    }
}

File: core/src/main/kotlin/com/yopox/TitleScreen.kt
package com.yopox

import com.badlogic.gdx.Input
import com.badlogic.gdx.graphics.Texture
import com.yopox.ui.draw
import com.yopox.ui.toLetterList
import ktx.graphics.use
import ktx.math.vec2

private const val OPTIONS = 3

class TitleScreen : AbstractScreen() {
    private val background = Texture("title.png")
    private val classique = "Partie classique".toLetterList()
    private val seed = "Partie avec seed".toLetterList()
    private val option = "Troisième option".toLetterList()
    private val arrow = ">".toLetterList()

    private var selected = 0

    override fun render(delta: Float) {
        super.render(delta)

        batch.use {
            it.draw(background, 0f, 0f)
            classique.draw(vec2(TILE * 15, TILE * 12), batch)
            seed.draw(vec2(TILE * 15, TILE * 10), batch)
            option.draw(vec2(TILE * 15, TILE * 8), batch)
            arrow.draw(vec2(TILE * 13, TILE * 12 - 2 * TILE * selected), batch)
        }
    }

    override fun keyUp(keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.UP -> selected = (selected - 1 + OPTIONS) % OPTIONS
            Input.Keys.DOWN -> selected = (selected + 1 + OPTIONS) % OPTIONS
            else -> Nupes.setScreen<GameScreen>()
        }
        return true
    }

    override fun keyTyped(character: Char): Boolean {
        return false
    }
}